configuration/src/lib.rs

```diff
@@ -135,44 +136,55 @@ impl NomadConfig {
-                        network,
-                    );
-                }
+                let deploy_info = self.core.get(connection);
+                eyre::ensure!(
+                    deploy_info.is_some(),
+                    "Missing core for {}, which is connected to {}.",
+                    connection,
+                    network,
+                );
+
+                eyre::ensure!(
+                    deploy_info.unwrap().has_replica(network),
+                    "Replica named '{}' not present on core named '{}' despite being listed in core connections",
+                    network,
+                    connection,
+                );
             }
         }
 
         // Check each core contains replicas ONLY for its listed connections
-        for (name, network) in self.core.iter() {
-            eyre::ensure!(
-                self.networks.contains(name),
-                "Core named '{}' not present in configured networks",
-                name,
-            );
-
-            // Check each replica
-            for replica in network.replicas() {
-                // Check that the network is known
+        for (name, deploy_info) in self.core.iter() {
+            if let CoreDeploymentInfo::Substrate(_) = deploy_info {
+                // No such thing as replicas ON substrate chain currently
+                continue;
+            } else {
                 eyre::ensure!(
                     self.networks.contains(name),
-                    "Replica named '{}' on core named '{}' not present in base config's configured networks",
-                    replica,
+                    "Core named '{}' not present in configured networks",
                     name,
                 );
-                // Check for replicas found, but not configured
-                eyre::ensure!(
-                    self.protocol.networks.get(name).unwrap().connections.contains(replica),
-                    "Replica named '{}' on core named '{}' not present in core's configured connections",
-                    replica,
-                    name
-                );
+
+                // Check each replica
+                for replica in deploy_info.replicas() {
+                    // Check that the network is known
+                    eyre::ensure!(
+                        self.networks.contains(replica),
+                        "Replica named '{}' on core named '{}' not present in base config's configured networks",
+                        replica,
+                        name,
+                    );
+                    // Check for replicas found, but not configured
+                    eyre::ensure!(
+                        self.protocol.networks.get(replica).unwrap().connections.contains(name),
+                        "Replica named '{}' on core named '{}' not present in core's configured connections",
+                        replica,
+                        name
+                    );
```

Comment:
Compare L181-186 to previous statement (which I deleted) below. Believe this was a bug fix in our semantics. `connections` means **outbound connections** but previous code treated this as _inbound_ connections. Want second eye @lattejed @prestwich 

Before:
```
eyre::ensure!(
   self.protocol.networks.get(name).unwrap().connections.contains(replica),
   "Replica named '{}' on core named '{}' not present in core's configured connections",
   replica,
   name
);
```

After:
```
// Check for replicas found, but not configured
eyre::ensure!(
    self.protocol.networks.get(replica).unwrap().connections.contains(name),
    "Replica named '{}' on core named '{}' not present in core's configured connections",
    replica,
    name
);
```

Reply:
I'd say the change to `self.networks.contains(replica)` at L175 is correct. Previously it was just repeatedly re-checking the presence of `name` 

For your change at L182, I'm pretty sure the original was correct. We're just checking if `network` of `name` has `replica` in its list of `connections`. 

However I think `// Check for replicas found, but not configured` is unclear 

And I'm not sure if `// Check each core contains replicas ONLY for its listed connections` maps to the actual checks we're doing in that section of the validation.

Reply:
Actually, I think you're correct on the second part. Reasoning:

We're matching each core.replica.<network> to a protocol.<network>.connection and checking for the presence of name. Your inbound vs outbound correction holds imo

Reply:
Cool so they are semantically correct but hard to parse. I will document these checks better, tbh I found them equally hard to parse when making the changes